### ISSUE TITLE
Replace ElasticSearch with Embedded Derby

### ISSUE THREAD

**Issue by jhmartin** [armzilla/amazon-echo-ha-bridge#76]
ElasticSearch is a very heavy method to run on a Pi just to store a few configuration values. This issue is to replace it with Apache Derby.

**Comment by armzilla** [comment]
i agree.  Some can argue that derby is heavyweight as well.  Others have proposed something as simple as storing the configuration as json on disk.  What do you think?

**Comment by jhmartin** [comment]
On-disk JSON would be fine too, I've updated the issue to indicate such.

**Comment by digitalroadies** [comment]
Might want to consider SQLite as well.

**Comment by linuxkidd** [comment]
Agreed.. Anything to reduce system load would be great.  I'm running this on a Pi3, and it takes 13 MINUTES from start to ports open and ready to work.

**Comment by armzilla** [comment]
sorry all, my day yob is taking up most if not all of my time.  If someone
has time to do this i can accept the pull, else please be patient!

On Thu, Jul 7, 2016 at 6:50 PM, Michael J. Kidd <notifications@github.com>
wrote:

> Agreed.. Anything to reduce system load would be great. I'm running this
> on a Pi3, and it takes 13 MINUTES from start to ports open and ready to
> work.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/armzilla/amazon-echo-ha-bridge/issues/76#issuecomment-231240578>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABTxZoIN7vQdCmnmnfegXeJyp-aP8Cs8ks5qTZDMgaJpZM4IkB-f>
> .
>



-- 
chicken goes cluck cluck, cow goes moo, lemur goes phh phh how bout you?

**Comment by linuxkidd** [comment]
Hey, no worries Arm... I know I certainly understand life getting in the way.

**Comment by pvdh** [comment]
For the record: startup on my raspberry 2 took about 250s, so I tried to replace ElasticSearch mit h2 and hibernate. Got me down to 240s. You can find my attempt at [pvdh/amazon-echo-ha-bridge](https://github.com/pvdh/amazon-echo-ha-bridge).
Doesn't seem to be worth the effort. Perhaps better results can be achieved by accessing a standalone database server, instead of an embedded one, perhaps ditching hibernate might be effective as well. However, I don't expect people running the bridge on a raspberry to have a database sever nearby, so I don't think it's worth exploring.

**Comment by jhmartin** [comment]
@pvdh 10s doesn't sound worth that effort, but the configuration is simple enough that ditching the database entirely for a simple json file should be sufficient.
